paasta_tools/kubernetes_tools.py

```diff
@@ -1246,3 +1329,109 @@ def fromstring(cls, _str: str) -> int:
 
 def is_kubernetes_available() -> bool:
     return Path(KUBE_CONFIG_PATH).exists()
+
+
+def create_secret(
+    kube_client: KubeClient,
+    secret: str,
+    service: str,
+    secret_provider: BaseSecretProvider,
+) -> None:
+    service = sanitise_service_name(service)
+    kube_client.core.create_namespaced_secret(
+        namespace="paasta",
+        body=V1Secret(
+            metadata=V1ObjectMeta(
+                name=f"paasta-secret-{service}-{secret}",
+                labels={"yelp.com/paasta_service": service},
+            ),
+            data={secret: base64.b64encode(secret_provider.decrypt_secret_raw(secret)).decode('utf-8')},
+        ),
+    )
+
+
+def update_secret(
+    kube_client: KubeClient,
+    secret: str,
+    service: str,
+    secret_provider: BaseSecretProvider,
+) -> None:
+    service = sanitise_service_name(service)
+    kube_client.core.replace_namespaced_secret(
+        name=f"paasta-secret-{service}-{secret}",
+        namespace="paasta",
```

Comment:
are we going to have to do some rbac bits to allow operators in other ns to get this?

Reply:
That's a good question. I just found `They can only be referenced by pods in that same namespace` as a snippet in the docs so I guess initially we can't with this code...

I guess we'll have to parametrise this and sync to multiple namespaces...

Maybe we could even have a map of `k8s namespace` =>`[]services` and `paasta` => `*` but other more specific namespaces would only sync a few services secrets?

I'll probably tackle this as a follow up ticket if you don't object